Assets/_Laz/Editor/PatrolLocationHandler.cs

```diff
@@ -0,0 +1,123 @@
+using System.Collections.Generic;
+using UnityEditor;
+using Laz;
+using UnityEngine;
+
+
+namespace LazEditor
+{
+    [CustomEditor(typeof(PatrolBehaviour))]
+    public class PatrolLocationHandler : Editor
+    {
+        private const float DRAW_PLANE_HEIGHT = 0f;
+        private const float CLOSEST_DISTANCE_FOR_DELETION = 0.6f;
+        private PatrolBehaviour _behaviour = null;
```

Comment:
So because you're using PatrolBehaviour directly, it'll now become tightly coupled. 
So if someone messes with PatrolBehaviour, this might break, and we don't want that. 

What you should do, is to create an Interface, IPatrolBehaviour and with methods that you need instead of manipulating `PatrolBehaviour.PatrolPosition` Directly.

So that if we ever have a AI, bot, moving obstacles that needs this, It just needs to Implement IPatrolBehaviour, it'll be good.

Reply:
So I added the Interface in my latest commit  (46b1f08) so that all inheritors need to have the Property. But I don't personally see the body needing to ever be different? Maybe I should use an abstract class instead of an interface and add the logic there?

What do you think?

Reply:
mmm i would stick with Interface.. 
cuz  if we have  FlyingRobot() Class, which derives from BaseMonster() Abstract class, then it won't be able to derive from patrol.

Reply:
and in terms of understanding.

A planetoid shoulnd't derive from patroling. 

like it's not part of a "patrol" object. It's a planetoid that so happens to be able to patrol

Reply:
Sounds good, so are there any more concerns with the PR or is it ready to be approved @m4rtint ?

Reply:
you should change this type to IPatrolBehaviour.

Reply:
And also rename behaviour to patrol Behaviour. So I have context.

Reply:
So turns out you cant make a custom editor for Interfaces, only Monobehaviour and ScriptableObject derived classes https://answers.unity.com/questions/1247734/custom-editor-for-interfacechild-classes-not-displ.html

So either we make a custom editor for each patrolling class (which seems bad) or we create an abstract patrolling class that each new patrol class derives from (attach as a different script)

So for example the flying robot gameObject has 2 different scripts, The flying robot class which derived from BaseMonster. and a second script, FlyingPatrol that derives from the new PatrolBehaviour abstract class

What do you think @m4rtint ?

Reply:
Sorry i think you misunderstood. 
`private PatrolBehaviour _behaviour = null;`

this line change it to 
`private IPatrolBehaviour _behaviour = null;`

Reply:
and then update the Interface to add
```
public Vector3 Position
        {
            get;
        }
        
        public Object GetUnityObject
        {
            get;
        }
```